edi_purchase_oca/tests/test_edi_configuration.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 CamptoCamp SA,
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import mock
+
+from odoo.exceptions import UserError
+
+from odoo.addons.edi_oca.tests.common import EDIBackendCommonComponentRegistryTestCase
+
+
+class TestsPurchaseEDIConfiguration(EDIBackendCommonComponentRegistryTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls._load_module_components(cls, "edi_purchase_oca")
+        cls.edi_configuration = cls.env["edi.configuration"]
+        cls.purchase_order = cls.env["purchase.order"]
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Product 1",
+                "default_code": "1234567",
+            }
+        )
+
+    def setUp(self):
+        super().setUp()
+        self.create_config = self.edi_configuration.create(
+            {
+                "name": "Create Config",
+                "active": True,
+                "code": "create_config",
+                "backend_id": self.backend.id,
+                "type_id": self.exchange_type_out.id,
+                "trigger": "on_record_create",
```

Comment:
on_button_confirm_purchase_order is probably more significant to test in that module